File: backend/internal/models/family_group.go
package models

import "time"

type FamilyGroup struct {
	ID        int64     `json:"id" db:"id"`
	Name      string    `json:"name" db:"name"`
	OwnerID   int64     `json:"owner_id" db:"owner_id"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
}

type FamilyMember struct {
	GroupID int64  `json:"group_id" db:"group_id"`
	UserID  int64  `json:"user_id" db:"user_id"`
	Role    string `json:"role" db:"role"`
}

type CreateFamilyGroupRequest struct {
	Name string `json:"name" binding:"required"`
}

type AddFamilyMemberRequest struct {
	UserID int64  `json:"user_id" binding:"required"`
	Role   string `json:"role" binding:"required"`
}

File: app/lib/models/family_group.dart
class FamilyGroup {
  final int id;
  final String name;
  final int ownerId;
  final DateTime createdAt;

  FamilyGroup({
    required this.id,
    required this.name,
    required this.ownerId,
    required this.createdAt,
  });

  factory FamilyGroup.fromJson(Map<String, dynamic> json) {
    return FamilyGroup(
      id: json['id'] as int,
      name: json['name'] as String,
      ownerId: json['owner_id'] as int,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }
}

class FamilyMember {
  final int groupId;
  final int userId;
  final String role;

  FamilyMember({
    required this.groupId,
    required this.userId,
    required this.role,
  });

  factory FamilyMember.fromJson(Map<String, dynamic> json) {
    return FamilyMember(
      groupId: json['group_id'] as int,
      userId: json['user_id'] as int,
      role: json['role'] as String,
    );
  }
}
